types/V1.mli

```diff
@@ -190,6 +190,22 @@ module type FLOW = sig
       before returning. At this point no data can flow in either direction
       and resources associated with the flow can be freed.
       *)
+
+  val disconnect: flow -> unit io
```

Comment:
My main objective is to clean up the resources rather than leak ... so I don't mind whether it's `unit io` or `unit`. I suggested `unit io` to offer callers the choice of waiting for the resource cleanup or not but `unit` would also work.

If we go for `unit`: would we require implementations of `connect` to catch any out-of-resources errors and wait for pending cleanups or would we pass the errors to the application and require the application to sleep and retry?